compose/cli/main.py

```diff
@@ -563,8 +563,14 @@ def images(self, options):
             rows = []
             for container in containers:
                 image_config = container.image_config
+                service = self.project.get_service(container.service)
+                if service.image_name in image_config['RepoTags']:
```

Comment:
Because each `RepoTags` entry is an explicit image+tag identifier (e.g. `img_name:tag_name`), there's a few scenarios where this condition fails:
- Any service where the image the result of a `build` section
- Any service declaring `image: foo` instead of `image: foo:latest`

Maybe it's worth appending a `:latest` tag if none is present?